### PR TITLE
feat: initial `firebase_app_check` preview release

### PR DESCRIPTION
**Pull request by Salakar**
## Description

**TODO**
 - [x] Add CI workflow config (incl. `test_driver`)
 - [x] Update package descriptions and readmes & ensure all package versions are 0.0.1
 - [x] Add app-check documentation section & overview page at https://firebase.flutter.dev/docs/app-check/overview
 	- [ ] Document installing debug provider for:
 		- [ ] Web
 		- [ ] iOS,macOS
 		- [ ] Android
 - [x] Add to overview table on https://firebase.flutter.dev/ - under a `Preview` sub section header.
 - [ ] Final pass of testing using the Debug provider docs added from previous todos. 

**Post release checklist**

 - [ ] Toggle visibility of `firebase_app_check_platform_interface` and `firebase_app_check_web` packages on pub.dev

## Related Issues

N/A

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).
This will ensure a smooth and quick review process. Updating the `pubspec.yaml` and changelogs is not required.

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [ ] All existing and new tests are passing.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`melos run analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I signed the [CLA].
- [ ] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change.
- [ ] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by Salakar** [comment]
@googlebot I consent.

**Comment by Salakar** [comment]
cc @Ehesp for `@googlebot I consent.`

**Comment by Ehesp** [comment]
@googlebot I consent.

File: packages/firebase_app_check/firebase_app_check/lib/src/firebase_app_check.dart [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  static FirebaseAppCheck get instance {
+    _instance ??= FirebaseAppCheck._(app: Firebase.app());
+    return _instance!;
+  }
+
+  /// Activates the Firebase App Check service.
+  ///
+  /// On web, provide the reCAPTCHA v3 Site Key which can be found in the
+  /// Firebase Console. For more information, see
+  /// [the Firebase Documentation](https://firebase.google.com/docs/app-check/web?authuser=0).
```
Comment (mikehardy) [inline packages/firebase_app_check/firebase_app_check/lib/src/firebase_app_check.dart:44]: `authuser=0` does not appear necessary in my testing

```suggestion
  /// [the Firebase Documentation](https://firebase.google.com/docs/app-check/web).
```

File: packages/firebase_app_check/firebase_app_check_platform_interface/lib/src/platform_interface/platform_interface_firebase_app_check.dart [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+import 'package:plugin_platform_interface/plugin_platform_interface.dart';
+import '../method_channel/method_channel_firebase_app_check.dart';
+
+abstract class FirebaseAppCheckPlatform extends PlatformInterface {
+  /// The [FirebaseApp] this instance was initialized with.
+  FirebaseAppCheckPlatform({this.appInstance}) : super(token: _token);
+
+  /// Create an instance using [app] using the existing implementation
+  factory FirebaseAppCheckPlatform.instanceFor({required FirebaseApp app}) {
+    // Only the default app is supported on App Check.
```
Comment (mikehardy) [inline packages/firebase_app_check/firebase_app_check_platform_interface/lib/src/platform_interface/platform_interface_firebase_app_check.dart:16]: is that true? ios documentation at least implies secondary apps are supported https://firebase.google.com/docs/reference/ios/firebaseappcheck/api/reference/Classes

same on android https://firebase.google.com/docs/reference/android/com/google/firebase/appcheck/FirebaseAppCheck#public-method-summary

Comment (Salakar) [reply packages/firebase_app_check/firebase_app_check_platform_interface/lib/src/platform_interface/platform_interface_firebase_app_check.dart:16]: With the current API there's some limitations in supporting this cross platform (some major underlying SDK differences making things difficult) so for the initial implementation of App Check in FlutterFire it won't be supported.

**Comment by Salakar** [comment]
Closed in favour of #6173 (to get website preview builds working)

**Pull request closed without merging**